File: VB/AttendanceRepository.vb
Imports Attendance.AttendanceBusiness
Imports Framework.UI

Public Class AttendanceRepository
    Inherits AttendanceRepositoryBase
    Private _isAvailable As Boolean
    Public Function GetComboboxData(ByRef cbxData As ComboBoxDataDTO) As Boolean
        Using rep As New AttendanceBusinessClient
            Try
                Return rep.GetComboboxData(cbxData)
            Catch ex As Exception
                rep.Abort()
                Throw ex
            End Try
        End Using
    End Function
End Class

File: C#/AttendanceRepository.cs
using System;
using Attendance.AttendanceBusiness;
using Framework.UI;

public class AttendanceRepository : AttendanceRepositoryBase
{
    private bool _isAvailable;
    public bool GetComboboxData(ref ComboBoxDataDTO cbxData)
    {
        using (AttendanceBusinessClient rep = new AttendanceBusinessClient())
        {
            try
            {
                return rep.GetComboboxData(cbxData);
            }
            catch (Exception ex)
            {
                rep.Abort();
                throw ex;
            }
        }
    }
}
